Gifski/ExportModifiedVideo.swift

```diff
@@ -0,0 +1,323 @@
+import Foundation
+import AVKit
+import SwiftUI
+
+struct ExportModifiedVideo: View {
+	@Environment(AppState.self) private var appState
+	@Environment(\.dismiss) private var dismiss
+	let input: Self.Input
+	@State private var state = ConvertState.generating(progress: 0)
+
+	var body: some View {
+		VStack(spacing: 20) {
+			switch state {
+			case let .empty(error: error):
+				if let error {
+					Text(error)
+				}
+				Button("Cancel") {
+					dismiss()
+				}
+			case .generating, .finished:
+				ProgressView()
+				if input.gifDuration >= 20 {
+					Button("Cancel") {
+						dismiss()
+					}
+				}
+			}
+		}
+		.padding()
+		.task(priority: .medium) {
+			do {
+				let task = Self.exportModifiedVideo(input: input)
+				self.state = .generating(progress: 0)
```

Comment:
It's safer to set state first.

```suggestion
				self.state = .generating(progress: 0)
				let task = Self.exportModifiedVideo(input: input)
```